Re-import html/semantics/popovers WPT

rwlbuis opened the pull request:
#### c7c487a14b4d9e1b5841ff093bb4838e7431e729
<pre>
Re-import html/semantics/popovers WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=257423">https://bugs.webkit.org/show_bug.cgi?id=257423</a>

Reviewed by Tim Nguyen.

Import based on 4c27189ed2.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-anchor-transition.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-anchor-transition.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-all-elements.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-basic.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-change-type.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-css-properties.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-css-properties.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-events.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-2.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-child-dialog-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-child-dialog.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hint-crash.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hide.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hide.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hover.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hover.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-show.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-show.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-toggle.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-toggle.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-light-dismiss.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-move-documents.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-not-keyboard-focusable.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display.tentative-expected.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display.tentative.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display.html.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-overlay-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-overlay.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-target-action-hover.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-target-action-hover.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-types-with-hints.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-types-with-hints.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-types.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/resources/popover-hover-hide-common.js: Added.
(async initialPopoverShow):
(async promise_test):
(runHoverHideTest):
(async runHoverHideTestsForInvokerAction.async string_appeared_here):
(async runHoverHideTestsForInvokerAction):
(makeTestParts):
(getComputedStyleTimeMs):
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/resources/popover-utils.js:
(isTopLayer):
(mouseOver):
(assertMouseStillOver):
(async mouseHover):
(msSinceMouseOver): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/w3c-import.log:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/popovers/popover-anchor-transition.tentative-expected.txt: Added.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-basic-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/popovers/popover-not-keyboard-focusable-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-expected.txt:
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/264649@main">https://commits.webkit.org/264649@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2644c5acbcb6b286372cb427ca156681e29ee31

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8262 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8554 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8771 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9929 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8326 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10542 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8465 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11200 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8407 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/9467 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7494 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10073 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6766 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7558 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/15118 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7889 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7695 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11047 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/8166 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6645 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7456 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/7470 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1985 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11665 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7909 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/platform/mac/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hint-crash.tentative-expected.txt:
Can we just skip the crash test and avoid generating the render tree dump?

rwlbuis replied:
Sure. Note, I suspect this is only happening because of the non standard crash test naming (probably the tentative part).

nt1m replied:
This is probably fixable in the test runner: https://searchfox.org/wubkat/search?q=crash.html&path=Tools%2F&case=false&regexp=false

(Separate from this commit though)

rwlbuis replied:
Yeah, I think I found the problem:
https://github.com/WebKit/WebKit/pull/14448

rwlbuis committed:
Re-import html/semantics/popovers WPT
https://bugs.webkit.org/show_bug.cgi?id=257423

Reviewed by Tim Nguyen.

Import based on 4c27189ed2.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-anchor-transition.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-anchor-transition.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-all-elements.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-basic.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-change-type.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-css-properties.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-css-properties.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-events.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-2.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-child-dialog-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-child-dialog.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hint-crash.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hide.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hide.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hover.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-hover.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-show.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-show.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-toggle.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-hover-hide-toggle.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-light-dismiss.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-move-documents.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-not-keyboard-focusable.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display.tentative-expected.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display.tentative.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-open-overflow-display.html.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-overlay-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-overlay.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-target-action-hover.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-target-action-hover.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-types-with-hints.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-types-with-hints.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/popover-types.html:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/resources/popover-hover-hide-common.js: Added.
(async initialPopoverShow):
(async promise_test):
(runHoverHideTest):
(async runHoverHideTestsForInvokerAction.async string_appeared_here):
(async runHoverHideTestsForInvokerAction):
(makeTestParts):
(getComputedStyleTimeMs):
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/resources/popover-utils.js:
(isTopLayer):
(mouseOver):
(assertMouseStillOver):
(async mouseHover):
(msSinceMouseOver): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/w3c-import.log:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/popovers/popover-anchor-transition.tentative-expected.txt: Added.
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/popovers/popover-attribute-basic-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/semantics/popovers/popover-not-keyboard-focusable-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/popovers/popover-focus-expected.txt:
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/264649@main
